File: Sys.java
import java.util.Scanner;
import java.time.LocalDate;
public class Sys {
    private final String password = "Z0O/1";
    static Scanner sc = new Scanner(System.in);
    public void system(Zoo zoo){
        int c = 0;
        String pass = "";
        System.out.println("===== BIENVENIDO AL SISTEMA DEL ZOOLOGICO =====\n");
        System.out.println("Ingrese la contraseña del sistema: ");
        pass = sc.nextLine();
        while (!pass.equals(password)) {
            c++;
            if (c == 5){
                System.out.println("Ha ingresado demasiadas veces la contraseña incorrecta...");
                return;
            }
            System.out.println("\nCONTRASEÑA INCORRECTA INTENTE DE NUEVO\n");
            System.out.println("Ingrese la contraseña del sistema: ");
            pass = sc.nextLine();
        }
        System.out.println("Contraseña correcta!");
        System.out.println("\n=== Bienvenido al Sistema!! ===\n");
        firstMenu(zoo);
    }

    private void firstMenu(Zoo zoo){
        boolean flag = true;
        while (flag) {
            System.out.println("\n\t== Que desea realizar: ==");
            System.out.println("1. Registrar");
            System.out.println("2. Revisar");
            System.out.println("3. Dirigir");
            System.out.println("4. Modificar");
            System.out.println("5. Cerrar programa");
            int opt = sc.nextInt();

            switch (opt) {
                case 1:
                    register(zoo);
                    break;
                case 2:
                    data(zoo);
                    break;
                case 3:
                    directions(zoo);
                    break;
                case 4:
                    edit(zoo);
                    break;
                case 5:
                    System.out.println("\nGRACIAS POR UTILIZAR EL SISTEMA!\n");
                    flag = false;
                    break;
                default:
                    System.out.println("\nHa ingresado un numero incorrecto!!\n");
                    break;
            }
        }
    }

    private void register(Zoo zoo){
        boolean flag = true;

        System.out.println("\n\tHA SELECCIONADO REGISTRAR\n");

        while(flag) {
            System.out.println("\t\n== Que desea registrar: ==");
            System.out.println("1. Nuevo Empleado");
            System.out.println("2. Nuevo Animal");
            System.out.println("3. Nuevo Cliente");
            System.out.println("4. Registrar Visita");
            System.out.println("5. Volver al menu");

            int opt = sc.nextInt();

            switch (opt) {
                case 1:
                    zoo.registerEmployee();
                    break;
                case 2:
                    zoo.registerAnimal();
                    break;
                case 3:
                    zoo.registerClient();
                    break;
                case 4:
                    zoo.registerVisit();
                    break;
                case 5:
                    flag = false;
                    break;
                default:
                    System.out.println("\nHa ingresado un numero no valido!!\n");
                    break;
            }
        }
    }

    private void data(Zoo zoo){
        boolean flag = true;

        System.out.println("\n\tHA SELECCIONADO REVISAR LOS DATOS DEL ZOO\n");
        while (flag){
            System.out.println("\t\n== Que desea consultar: ==");
            System.out.println("1. Empleados");
            System.out.println("2. Visitantes");
            System.out.println("3. Animales");
            System.out.println("4. Visitas");
            System.out.println("5. Asignacion de Empleados");
            System.out.println("6. Volver al menu");

            int opt = sc.nextInt();

            switch (opt){
                case 1:
                    zoo.printEmployees();
                    break;
                case 2:
                    zoo.printClients();
                    break;
                case 3:
                    zoo.printAnimals();
                    break;
                case 4:
                    zoo.printVisits();
                    break;
                case 5:
                    zoo.printOrders();
                    break;
                case 6:
                    flag = false;
                    break;
                default:
                    System.out.println("\nHa ingresado un numero no valido!!\n");
                    break;
            }
        }
    }

    private void directions(Zoo zoo){
        System.out.println("\n\tHA SELECCIONADO DIRIGIR A LOS EMPLEADOS\n");
        zoo.orderMaintenance();
    }

    private void edit(Zoo zoo){
        boolean flag = true;
        System.out.println("\n\tHA SELECCIONADO MODIFICAR EL SISTEMA\n");

        while (flag) {
            System.out.println("\nDesea eliminar o modificar algo del sistema: ");
            System.out.println("1. Modificar");
            System.out.println("2. Eliminar");
            System.out.println("3. Volver al menu");

            int opt = sc.nextInt();

            switch (opt) {
                case 1:
                    modify(zoo);
                    break;
                case 2:
                    delete(zoo);
                    break;
                case 3:
                    flag = false;
                    break;
                default:
                    break;
            }
        }
    }

    private void modify(Zoo zoo) {
        boolean flag = true;

        while (flag) {
            System.out.println("\nQue desea modificar: ");
            System.out.println("1. Cliente");
            System.out.println("2. Empleado");
            System.out.println("3. Animal");
            System.out.println("4. Volver");

            int opt = sc.nextInt();

            switch (opt) {
                case 1 -> zoo.modifyClient();
                case 2 -> zoo.modifyEmployee();
                case 3 -> zoo.modifyAnimal();
                case 4 -> flag = false;
                default -> System.out.println("Opcion ingresada incorrecta");
            }
        }
    }

    private void delete(Zoo zoo){
        boolean flag = true;

        while (flag) {
            System.out.println("\nQue desea eliminar: ");
            System.out.println("1. Cliente");
            System.out.println("2. Empleado");
            System.out.println("3. Animal");
            System.out.println("4. Volver");

            int opt = sc.nextInt();

            switch (opt) {
                case 1 -> zoo.deleteClient();
                case 2 -> zoo.deleteEmployee();
                case 3 -> zoo.deleteAnimal();
                case 4 -> flag = false;
                default -> System.out.println("Opcion ingresada incorrecta");
            }
        }
    }
}

File: Sys.py
import getpass
import Zoo
class Systm:

    Zoo.Zoologic.employees
    Zoo.Zoologic.animals
    Zoo.Zoologic.clients
    Zoo.Zoologic.orders
    Zoo.Zoologic.visits

    password="ZOO/1"

    def system(self):
        c=0
        print("=====BIENVENIDO AL SISTEMA DEL ZOOLOGICO=====")
        print("Ingrese la contraseña del sistema")
        pass_input =input()
        while pass_input != self.password:
            c+=1
            if c==5:
                print("Demasiados intentos fallidos")
                break
            else:
                print("Contraseña incorrecta, intente de nuevo")
                pass_input= getpass.getpass("Ingresa la contraseña del sistema")

        print("Contraseña correcta, bienvenido!!")
        self.first_menu()

    def first_menu(self):
        flag= True
        while flag:
            print("==Que accion quiere realizar?==")
            print("1. Registrar")
            print("2. Consultar")
            print("3. Dirigir")
            print("4. Modificar")
            print("5. Cerrar programa")
            opt=int(input())

            if opt==1:
                self.register()
            elif opt==2:
                self.check()
            elif opt==3:
                self.manage()
            elif opt==4:
                self.modify()
            elif opt==5:
                flag= False
            else:
                print("Has ingresado un numero invalido")

    def register(self):
        flag=True
        print ("==SELLECIONASTE: REGISTRAR==")

        while flag:
            print("Que quieres registrar?")
            print ("1. Nuevo empleado")
            print("2. Nuevo animal")
            print("3. Nuevo cliente")
            print("4. Registrar visita")
            print("5. Regresar al menu")

            opt=int(input())

            if opt==1:
                Zoo.Zoologic.register_employee(self)
            elif opt==2:
                Zoo.Zoologic.register_animal(self)
            elif opt==3:
                Zoo.Zoologic.register_client(self)
            elif opt==4:
                Zoo.Zoologic.register_visit(self)
            elif opt==5:
                flag= False
            else:
                print(" Ingresaste un numero invalido")

    def check(self):
        flag =True

        print ("==SELLECIONASTE: REVISAR==")

        while flag:
            print("Que quieres revisar?")
            print ("1. Empleados")
            print("2. Animales")
            print("3. Visitantes")
            print("4. Visitas")
            print("5. Ordenes")
            print("6. Regresar al menu")

            opt=int(input())

            if opt==1:
                Zoo.Zoologic.print_employees(self)
            elif opt==2:
                Zoo.Zoologic.print_animals(self)
            elif opt==3:
                Zoo.Zoologic.print_clients(self)
            elif opt==4:
                Zoo.Zoologic.print_visits(self)
            elif opt==5:
                Zoo.Zoologic.print_orders(self)
            elif opt==6:
                flag= False
            else:
                print(" Ingresaste un numero invalido")

    def manage(self):
        print("==Seleccionaste: DIRIGIR==")
        Zoo.Zoologic.create_maintenance_order(self)

    def modify(zoologic):
        flag=True
        while flag:
            print("Que desea modificar: ")
            print("1. Cliente")
            print("2. Empleado")
            print("3. Animal")
            print("4. Volver al menu")

            opt=int(input())
            
            if opt==1:
                Zoo.Zoologic.modify_client(zoologic)
            elif opt==2:
                Zoo.Zoologic.modify_employee(zoologic)
            elif opt==3:
                Zoo.Zoologic.modify_animal(zoologic)
            elif opt==4:
                flag=False
            else:
                print("Opcion invalida")
                
    def delete(zoologic):
        flag=True
        while flag:
            print("Que desea eliminar: ")
            print("1. Cliente")
            print("2. Empleado")
            print("3. Animal")
            print("4. Volver al menu")
            opt=input()
            if opt==1:
                Zoo.Zoologic.delete_client(zoologic)
            elif opt==2:
                Zoo.Zoologic.delete_employee(zoologic)
            elif opt==3:
                Zoo.Zoologic.delete_animal(zoologic)
            elif opt==4:
                flag=False
            else:
                print("Opcion invalida")
    